mindmeld/text_preparation/text_preparation_pipeline.py

```diff
@@ -480,66 +480,83 @@ def create_from_app_path(app_path):
-            tokenizer (str): Class name of Tokenizer to use.
-            stemmer (str): Class name of Stemmer to use.
+            preprocessors (Tuple[Union[str, Preprocessor]]): List of preprocessor class
+                names.
+            regex_norm_rules (List[Dict]): List of regex normalization rules represented
+                as dictionaries. ({"pattern":<pattern>, "replacement":<replacement>})
+            normalizers (Tuple[Union[str, Preprocessor]]): List of normalizer class names.
+            tokenizer (Union[str, Tokenizer]): Class name of Tokenizer to use.
+            stemmer (Union[str, Stemmer]): Class name of Stemmer to use.
 
         Returns:
             TextPreparationPipeline: A TextPreparationPipeline class.
         """
-        preprocessors = (
-            [PreprocessorFactory.get_preprocessor(p) for p in preprocessors]
+
+        # Instantiate Preprocessors
+        instantiated_preprocessors = (
+            TextPreparationPipelineFactory._instantiate_pipeline_components(
+                Preprocessor, preprocessors, language
+            )
             if preprocessors
             else [NoOpPreprocessor()]
         )
 
+        # Update Regex Normalization Exception Characters as Specified in the Config
         if keep_special_chars:
-            RegexNormalizerRuleFactory.EXCEPTION_CHARS = keep_special_chars
+            RegexNormalizerRuleFactory.set_exception_chars(keep_special_chars)
 
-        normalizers = (
-            [NormalizerFactory.get_normalizer(n) for n in normalizers]
+        # Instantiate Normalizers
+        instantiated_normalizers = (
+            TextPreparationPipelineFactory._instantiate_pipeline_components(
+                Normalizer, normalizers, language
+            )
             if normalizers
             else [NoOpNormalizer()]
         )
 
+        # Instatiate Regex Norm Rules as Normalizer Classes
         if regex_norm_rules:
             regex_normalizers = RegexNormalizerRuleFactory.get_regex_normalizers(
                 regex_norm_rules
             )
             # Adds the regex normalizers as the first normalizers by default
-            normalizers = regex_normalizers + normalizers
+            instantiated_normalizers = regex_normalizers + instantiated_normalizers
 
-        tokenizer = (
-            TokenizerFactory.get_tokenizer(tokenizer, language)
+        # Instantiate Tokenizer
+        instantiated_tokenizer = (
+            TextPreparationPipelineFactory._instantiate_pipeline_components(
+                Tokenizer, tokenizer, language
+            )
             if tokenizer
             else TokenizerFactory.get_tokenizer_by_language(language)
         )
-        stemmer = (
-            StemmerFactory.get_stemmer(stemmer)
+
+        # Instantiate Stemmer
+        instantiated_stemmer = (
+            TextPreparationPipelineFactory._instantiate_pipeline_components(
+                Stemmer, stemmer, language
```

Comment:
language not needed

Reply:
Removed the language param input for preprocessors, normalizers, and stemmers